internal/teststep/config.go

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package teststep
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+const (
+	testCaseProviderConfigFileName = "test_case_provider_config.tf"
+	testStepProviderConfigFileName = "test_step_provider_config.tf"
+)
+
+var (
+	providerConfigBlockRegex  = regexp.MustCompile(`provider "?[a-zA-Z0-9_-]+"? {`)
+	terraformConfigBlockRegex = regexp.MustCompile(`terraform {`)
+)
+
+type Config interface {
```

Comment:
`HasDirectory()`, `HasRaw()`, `Raw()` and `WriteDirectory()` can all be removed and replaced with `Write(dest string)` once the logic for writing the raw configuration derived from `TestStep.Config` has been moved out of `*WorkingDir.SetConfig()`.